### PR TITLE
ESC-873 = feat: implement handling of special characters in select options and …

### PR DESCRIPTION
**Pull request by chiragchhatrala**
…add corresponding tests

- Removed the previous escaping logic for select and multi-select options.
- Added a new test suite to validate form submissions with special characters, including tabs, backslashes, newlines, unicode, commas, and quotes.
- Ensured that form submissions correctly handle and store these special characters.

### CODE REVIEW

File: api/tests/Feature/Forms/SelectOptionSpecialCharsTest.php [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                    ['id' => 'opt2', 'name' => 'Normal'],
+                ],
+            ],
+        ];
+
+        $this->form->properties = array_merge($this->form->properties, [$selectField]);
+        $this->form->save();
+
+        $formData = $this->generateFormSubmissionData($this->form, [
+            'select_by_id' => "Special\tOption — Опция",
```
Comment (JhumanJ) [inline api/tests/Feature/Forms/SelectOptionSpecialCharsTest.php]: This test says it submits the option by id, but the payload is the option name again. Please submit `opt1` here and assert the stored value, or rename the test so it does not claim id-based coverage.

**Review by JhumanJ — CHANGES_REQUESTED** [review]
Thanks for the focused fix here. I’m requesting changes because there is still a backslash edge case that this PR does not cover yet.

1. `AnswerFormRequest` still validates select values via `Rule::in($this->getSelectPropertyOptions($property))`. Laravel stringifies that rule and parses it with CSV/backslash escaping, so an option name that ends with a backslash is still rejected. I verified this directly in the app validator with an allowed value of `"Ends\\"`, which fails validation. Please cover and fix trailing-backslash option names, not only internal backslashes like `C:\\Users\\test`.

2. The “submit select option by id” test currently submits the option name again rather than `opt1`, so it does not exercise id-based submission. Please either rename the test or change the payload to submit the id and assert the stored value.
